### PR TITLE
DocumentDelete refactored

### PR DESCRIPTION
**Pull request by Phil91**
Fixed the document delete endpoint

### CODE REVIEW

File: src/administration/CatenaX.NetworkServices.Administration.Service/Controllers/DocumentsController.cs [hunk 1]
```diff
@@ -55,7 +55,7 @@ public async Task<ActionResult> GetDocumentContentFileAsync([FromQuery] Guid doc
     /// <response code="403">The user is not assigned with the Company.</response>
     /// <response code="404">The document was not found.</response>
     [HttpDelete]
-    [Authorize(Roles = "delete_documents")]
+    // [Authorize(Roles = "delete_documents")]
```
Comment (jjeroch) [inline src/administration/CatenaX.NetworkServices.Administration.Service/Controllers/DocumentsController.cs:58]: commented out the auth validation? That means not auth validation of the user permission

File: src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/Repositories/IDocumentRepository.cs [hunk 2]
```diff
@@ -51,4 +51,9 @@ public interface IDocumentRepository
     /// <returns>Returns the user id if a document is found for the given id, otherwise null</returns>
     Task<(Guid DocumentId, bool IsSameUser)> GetDocumentIdCompanyUserSameAsIamUserAsync(Guid documentId, string iamUserId);
 
+    /// <summary>
+    /// Removes the given document with from the database
```
Comment (jjeroch) [inline src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/Repositories/IDocumentRepository.cs:55]: very minor thing, but lets change the summary to "Deleting document record and document file from the portal db/document storage location"

**Review by jjeroch — CHANGES_REQUESTED** [review]

File: src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/Repositories/IDocumentRepository.cs (hunk 2 above)
Comment (Phil91) [reply src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/Repositories/IDocumentRepository.cs]: Changes the comment

File: src/administration/CatenaX.NetworkServices.Administration.Service/Controllers/DocumentsController.cs (hunk 1 above)
Comment (Phil91) [reply src/administration/CatenaX.NetworkServices.Administration.Service/Controllers/DocumentsController.cs]: back in, that shouldn't haved been pushed

**Review by jjeroch — APPROVED** [review]

**Comment by jjeroch** [comment]
ready to merge

**Pull request merged**